drivers/flash/nxp_qspi_nor.c

```diff
@@ -0,0 +1,419 @@
+
+	/* Exit QPI mode */
+	[4 * NOR_CMD_LUT_SEQ_IDX_EXITQPI] =
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_4PAD, 0xF5, kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0),
+
+	/* Erase whole chip */
+	[4 * NOR_CMD_LUT_SEQ_IDX_ERASECHIP] =
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_1PAD, 0xC7, kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0),
+};
+
+static status_t flash_mcux_flexspi_qspi_wait_bus_busy(struct device *dev, off_t offset)
+{
+	bool is_busy = true;
+	uint32_t read_value;
+	status_t status;
+	flexspi_transfer_t flash_transfer;
+
+	struct flash_priv *priv = dev->driver_data;
+	FLEXSPI_Type *base_address = priv->pflash_block_base;
+
+	flash_transfer.deviceAddress = offset;
+	flash_transfer.port = kFLEXSPI_PortA1;
+	flash_transfer.cmdType = kFLEXSPI_Read;
+	flash_transfer.SeqNumber = 1;
+	flash_transfer.seqIndex = NOR_CMD_LUT_SEQ_IDX_READSTATUS;
+	flash_transfer.data = &read_value;
+	flash_transfer.dataSize = 1;
+
+	do{
+		status = FLEXSPI_TransferBlocking(base_address, &flash_transfer);
+		if (status != kStatus_Success) {
+			return status;
+		}
+		// Status bit information for IS25LP064.
+		//
+		// The Write In Progress (WIP) bit is read-only, and can be used to
+		// detect the progress or completion of a program or erase operation. When the
+		// WIP bit is “0”, the device is ready for write status register, program or
+		// erase operation. When the WIP bit is “1”, the device is busy.
+		is_busy = ((read_value & 0x01UL) == 0x01UL) ? true : false;
+	} while (is_busy);
+
+	return status;
+}
+
+status_t flash_mcux_flexspi_qspi_write_enable(struct device *dev, off_t offset)
+{
+	status_t status;
+	flexspi_transfer_t flash_transfer;
+
+	struct flash_priv *priv = dev->driver_data;
+	FLEXSPI_Type *base_address = priv->pflash_block_base;
+
+	flash_transfer.deviceAddress = offset;
+	flash_transfer.port = kFLEXSPI_PortA1;
+	flash_transfer.cmdType = kFLEXSPI_Command;
+	flash_transfer.SeqNumber = 1;
+	flash_transfer.seqIndex = NOR_CMD_LUT_SEQ_IDX_WRITEENABLE;
+
+	status = FLEXSPI_TransferBlocking(base_address, &flash_transfer);
+
+	return status;
+}
+
+static int flash_mcux_flexspi_qspi_read(struct device *dev, off_t offset,
+					void *data, size_t len)
+{
+	status_t status;
+	flexspi_transfer_t flash_transfer;
+	unsigned int key;
+
+	key = irq_lock();
+
+	struct flash_priv *priv = dev->driver_data;
+	FLEXSPI_Type *base_address = priv->pflash_block_base;
+
+	flash_transfer.deviceAddress = offset;
+	flash_transfer.port = kFLEXSPI_PortA1;
+	flash_transfer.cmdType = kFLEXSPI_Read;
+	flash_transfer.SeqNumber = 1;
+	flash_transfer.seqIndex = NOR_CMD_LUT_SEQ_IDX_READ_FAST_QUAD;
+	flash_transfer.data = (uint32_t *)data;
+	flash_transfer.dataSize = len;
+
+	status = FLEXSPI_TransferBlocking(base_address, &flash_transfer);
+	if (status == kStatus_Success) {
+		status = flash_mcux_flexspi_qspi_wait_bus_busy(dev, offset);
+	}
+
+	irq_unlock(key);
+
+	return status;
+}
+
+#define FLASH_WRITE_SIZE 256
```

Comment:
it's better to be defined in device tree.

Reply:
Yes, but what node should it be part of?

Reply:
It should be added to the binding YAML for `nxp,imx-flexspi` as a property named something like `zephyr,qspi-ram-buffer-size` and documented per the pending material in #27624, specifically the edits to `d-vs-kconfig.rst` on how to use devicetree properties to inform drivers of instance-specific configuration data.  So it goes in the `&flexspi {}` node.

Reply:
Do you have an example of any other driver doing this? I don't exactly understand the details here.